src/stores/WalletHistoryStore.js

```diff
@@ -0,0 +1,63 @@
+import { observable, action } from 'mobx';
+import _ from 'lodash';
+
+import Transaction from './models/Transaction';
+import { queryAllTransactions } from '../network/graphQuery';
+import { SortBy, TransactionType } from '../constants';
+
+
+export default class WalletHistoryStore {
+  @observable fullList = []
+  @observable list = []
+  @observable orderBy = 'createdTime'
+  @observable direction = SortBy.Descending
+  @observable limit = 500
+  @observable skip = 0
+  @observable perPage = 10
+  @observable page = 0
+  @observable expanded = []
+
+  @action.bound
+  queryTransactions = async (orderBy = this.orderBy, direction = this.direction, limit = this.limit, skip = this.skip) => {
+    try {
+      const filters = [{ type: TransactionType.Transfer }];
+      const orderByObj = { field: orderBy, direction };
+
+      const result = await queryAllTransactions(filters, orderByObj, limit, skip);
+      this.fullList = _.map(result, (tx) => new Transaction(tx));
+      this.list = _.slice(this.fullList, 0, this.perPage);
+    } catch (error) {
+      console.error(error); // eslint-disable-line
+      this.fullList = [];
+      this.list = [];
+    }
+  }
+
+  @action.bound
+  onPageChange = (page) => {
+    this.expanded = [];
+    this.page = page;
+
+    const start = this.page * this.perPage;
+    this.list = _.slice(this.fullList, start, start + this.perPage);
```

Comment:
This is a great place for a `reaction`
WalletHistory/index.js
```js
          <TableFooter
              fullList={walletHistory.fullList}
              perPage={walletHistory.perPage}
              page={walletHistory.page}
              onPageChange={(page) => walletHistory.page = page}
              onPerPageChange={(perPage) => walletHistory.perPage = perPage}
            />
```
HistoryStore.js
```js
class HistoryStore {
  @observable fullList
  @observable page
  @observable perPage
  @observable list = []
  constructor() {
    reaction(
      () => this.perPage, // when perPage changes
      () => this.list = _.slice(this.fullList, 0, this.perPage);
    )
    reaction(
      () => this.page, // when page changes
      () => {
        const start = this.page * this.perPage;
        this.list = _.slice(this.fullList, start, start + this.perPage)
      }
    ) 
  }
}
```

Reply:
good call 👍